### PR TITLE
fix: make sure the code passes new ameba

### PR DESCRIPTION
**Pull request by anton7c3**

### CODE REVIEW

File: src/aws-credentials/provider.cr [hunk 1]
```diff
@@ -44,14 +44,14 @@ module Aws::Credentials
     end
 
     private def resolve_credentials : Credentials
-      @providers.find { |p|
-        p.credentials? ? true : false
+      @providers.find { |provider_|
```
Comment (y2k2mt) [inline src/aws-credentials/provider.cr]: [Q] why ends with underscore?

File: src/aws-credentials/http_client.cr [hunk 2]
```diff
@@ -7,8 +7,8 @@ module Aws::Credentials
       http = HTTP::Client.new uri
       begin
         http.connect_timeout = 5.seconds
-        headers.try do |h|
-          http.headers = h
+        headers.try do |hdr|
```
Comment (y2k2mt) [inline src/aws-credentials/http_client.cr]: I would prefer below.
```
headers -> maybe_headers
hdr -> headers
```

**Comment by y2k2mt** [comment]
Activate Github Actions

File: src/aws-credentials/provider.cr (hunk 1 above)
Comment (anton7c3) [reply src/aws-credentials/provider.cr]: Ameba forbids short names (e.g. `p`, `pp`), and I was out of creativity for a better name.

EDIT: after another look, I can collapse this block to the following:
```crystal
      @providers.find(&.credentials?).try &.credentials?
      || raise MissingCredentials.new "No provider serves credential : #{@providers.map(&.class.name)}"
```

File: src/aws-credentials/http_client.cr (hunk 2 above)
Comment (anton7c3) [reply src/aws-credentials/http_client.cr]: fixed

**Review by y2k2mt — APPROVED** [review]
LGTM

**Pull request merged**